Code/Framework/AzToolsFramework/AzToolsFramework/ComponentMode/ComponentModeActionHandler.cpp

```diff
@@ -100,6 +101,26 @@ namespace AzToolsFramework
                 }
             );
 
+            // Disable this action if the Entity Picker is enabled while in Component Mode.
+            m_actionManagerInterface->InstallEnabledStateCallback(
+                actionIdentifier,
```

Comment:
suggestion: It would be good to run this change by @mbalfour-amzn just to check this doesn't break anything with pick mode in the painting tool with terrain as it works slightly differently to the default entity pick mode. Should hopefully fine but would be good to verify with them. Thanks!

Reply:
This fix was added for that exact case. Pressing `Esc` to end Entity Picking mode would also end the Paintbrush Component Mode, while this change prevents that.

Reply:
Okay perfect! So this works for Component Mode use cases and normal pick mode when in the regular viewport?

Reply:
At the moment, this only impacts the "Done" action for Component Mode. The Behavior of the Entity Picker mode `Esc` interaction is still using the old "OnEscape" Notification bus call since that system has not been migrated yet and I didn't want to expand the scope of the changes and testing.

Reply:
I see got you, thanks for the explanation 👍